src/main/java/com/czertainly/core/messaging/listeners/NotificationListener.java

```diff
@@ -69,12 +139,82 @@ public void processMessage(NotificationMessage notificationMessage) {
                                         data.getApprovalUuid(), data.getObjectUuid(), data.getCreatorUsername(), data.getStatus().getLabel()),
                                 getApprovalNotificationDetail(data),
                                 notificationMessage);
+                        if (notificationTypeStringMap != null) {
+                            String notificationInstanceUUID = notificationTypeStringMap.get(NotificationType.APPROVAL_CLOSED);
+                            if (notificationInstanceUUID != null) {
+                                sendExternalNotifications(notificationInstanceUUID, notificationMessage, NotificationType.APPROVAL_CLOSED);
+                            }
+                        }
                     }
                 }
             }
         }
     }
 
+
+    private void sendExternalNotifications(String notificationInstanceUUID, NotificationMessage notificationMessage, NotificationType notificationType) {
+        Optional<NotificationInstanceReference> notificationInstanceReference = notificationInstanceReferenceRepository.findByUuid(UUID.fromString(notificationInstanceUUID));
+        if (notificationInstanceReference.isPresent()) {
```

Comment:
This if checking the notification instance reference will only skip if it is not available, but requested, see the warn message as part of the else.

What about something like:
```java
Optional<NotificationInstanceReference> notificationInstanceReference = notificationInstanceReferenceRepository.findByUuid(UUID.fromString(notificationInstanceUUID))
.orElseThrow(() -> new NotFoundException(NotificationInstanceReference.class, uuid));
```
?

Reply:
I tried to change it, but it can't resolve uuid in the exception